src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -452,6 +452,14 @@ public function newInstanceArgs(array|null $args = null): object
         throw new Exception\NotImplemented('Not implemented');
     }
 
+    /**
```

Comment:
Is it good idea to add method to adapter?

Reply:
Oh, true, this should only be on the BetterReflection API, **not** on the adapter

Reply:
ok, I wasn't able to differentiate which is what and thought I need to reflect the change there. even easier if this is not required :)

Reply:
I want to call this new method from [PHPStan ClassReflection](https://github.com/phpstan/phpstan-src/blob/1.11.x/src/Reflection/ClassReflection.php). I guess this means we need it on the adapter then?

Reply:
From an architectural PoV, it does not fit here: the adapter is only for mirroring the API of ext-reflection.

Anything beyond that already couples the consumer to BetterReflection, rather than ext-reflection, and should not be here.

Reply:
even then, remember that the dependency is to ext-reflection, and the adapters are not supposed to deviate from that API.